html/user/buda_submit.php

```diff
@@ -0,0 +1,351 @@
+    for ($i=0; $i<$n; $i++) {
+        $path = sprintf('%s/%s', $batch_dir, $batch_desc->shared_files[$i]);
+        [$md5, $size] = $batch_desc->shared_file_infos[$i];
+        $phys_name = sprintf('batch_%d_%s', $batch_id, $md5);
+        stage_file_aux($path, $md5, $size, $phys_name);
+        $batch_desc->shared_files_phys_names[] = $phys_name;
+    }
+    foreach ($batch_desc->jobs as $job) {
+        $n = count($batch_desc->unshared_files);
+        $job->phys_names = [];
+        for ($i=0; $i<$n; $i++) {
+            $path = sprintf('%s/%s/%s',
+                $batch_dir, $job->dir, $batch_desc->unshared_files[$i]
+            );
+            [$md5, $size] = $job->file_infos[$i];
+            $phys_name = sprintf('batch_%d_%s', $batch_id, $md5);
+            stage_file_aux($path, $md5, $size, $phys_name);
+            $job->phys_names[] = $phys_name;
+        }
+    }
+}
+
+function create_jobs(
+    $variant_desc, $batch_desc, $batch_id, $boinc_app, $batch_dir_name
+) {
+    // get list of names of app files
+    //
+    $app_file_names = $variant_desc->dockerfile_phys;
+    foreach ($variant_desc->app_files_phys as $pname) {
+        $app_file_names .= " $pname";
+    }
+    $job_cmds = '';
+    foreach ($batch_desc->jobs as $job) {
+        $job_cmd = sprintf('--wu_name batch_%d__job_%s', $batch_id, $job->dir);
+        if ($job->cmdline) {
+            $job_cmd .= sprintf(' --command_line "%s"', $job->cmdline);
+        }
+        $job_cmd .= " $app_file_names";
+        foreach ($batch_desc->shared_files_phys_names as $x) {
+            $job_cmd .= " $x";
+        }
+        foreach ($job->phys_names as $x) {
+            $job_cmd .= " $x";
+        }
+        $job_cmds .= "$job_cmd\n";
+    }
+    $cmd = sprintf(
+        'cd ../..; bin/create_work --appname %s --batch %d --stdin --command_line "--dockerfile %s --verbose" --wu_template %s --result_template %s',
+        $boinc_app->name, $batch_id, $variant_desc->dockerfile,
+        "buda_batches/$batch_dir_name/template_in",
+        "buda_batches/$batch_dir_name/template_out"
+    );
+    $cmd .= sprintf(' > %s 2<&1', "buda_batches/errfile");
+    $h = popen($cmd, "w");
+    if (!$h) error_page('create_work launch failed');
+    fwrite($h, $job_cmds);
+    $ret = pclose($h);
+    if ($ret) {
+        echo $cmd;
+        echo "\n\n";
+        echo $job_cmds;
+        echo "\n\n";
+        readfile("../../buda_batches/errfile");
+        exit;
+        error_page("create_work failed: $x");
+    }
+}
+
+///////////////// TEMPLATE CREATION //////////////
+
+function file_ref_in($fname) {
+    return(sprintf(
+'      <file_ref>
+          <open_name>%s</open_name>
+          <copy_file/>
+       </file_ref>
+',
+        $fname
+    ));
+}
+function file_info_out($i) {
+    return sprintf(
+'    <file_info>
+        <name><OUTFILE_%d/></name>
+        <generated_locally/>
+        <upload_when_present/>
+        <max_nbytes>5000000</max_nbytes>
```

Comment:
Of course it depends on the application type, but I assume 5MB is too few.

Reply:
At some point we'll need to let the job submitter specify this; 5MB is OK for now